Classes/Player/PlayerController.swift

```diff
@@ -53,12 +53,18 @@ class PlayerController: NSObject, Player {
         return self.currentPlayer.view
     }
     
-    public let sessionId = UUID()
+    public var sessionId: String {
+        return self.sessionUUID.uuidString + ":" + (self.mediaSessionUUID?.uuidString ?? "")
+    }
+    
+    let sessionUUID = UUID()
+    var mediaSessionUUID: UUID?
     
     public override init() {
         self.currentPlayer = AVPlayerEngine()
-        self.contentRequestAdapter = KalturaPlaybackRequestAdapter(playSessionId: self.sessionId)
         super.init()
+        // initial creation of play session id adapter
+        self.contentRequestAdapter = KalturaPlaybackRequestAdapter(sessionId: self.sessionId)
```

Comment:
This part should be done by PlayerLoader, not PlayerController. The reason: PlayerLoader handles all the plugins logic, and this is plugin-ish. It shouldn't be at the core.